pkg/termination/write_test.go

```diff
@@ -66,3 +66,29 @@ func TestExistingFile(t *testing.T) {
 		}
 	}
 }
+
+func TestMaxSizeFile(t *testing.T) {
+	value := ""
+	for n := 0; n < 4096; n++ {
+		value += "a"
+	}
+	tmpFile, err := ioutil.TempFile(os.TempDir(), "tempFile")
+	if err != nil {
+		log.Fatal("Cannot create temporary file", err)
+	}
+	// Remember to clean up the file afterwards
+	defer os.Remove(tmpFile.Name())
+
+	logger, _ := logging.NewLogger("", "test termination")
+	defer func() {
+		_ = logger.Sync()
+	}()
+	output := []v1alpha1.PipelineResourceResult{{
+		Key:   "key1",
+		Value: value,
+	}}
+
+	if err := WriteMessage(tmpFile.Name(), output); err == nil {
```

Comment:
Suggest checking for specific error kind here so that other errors we introduce by mistake are also caught:

```go
if err := WriteMessage(tmpFile.Name(), output); !errors.Is(err, MessageLengthError) {
  t.Fatalf("Expected MessageLengthError, receved: %v", err)
}
```

Reply:
Done, thanks